test/language/expressions/in/private-field-invalid-rhs.js

```diff
@@ -0,0 +1,26 @@
+// Copyright 2021 the V8 project authors.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Syntactic grammar restricts right-hand side
+info: |
+  Syntax
+    RelationalExpression[In, Yield, Await]:
+    [...]
+    [+In]PrivateIdentifier in ShiftExpression[?Yield, ?Await]
+esid: sec-relational-operators-runtime-semantics-evaluation
+negative:
+  phase: parse
+  type: SyntaxError
+features: [class-fields-private-in]
+---*/
+
+$DONOTEVALUATE();
+
+class C {
+  #field;
+
+  constructor() {
+    #field in () => {};
```

Comment:
this seems like it should parse just fine, but the test seems to be a negative one?

Reply:
I assumed that would be redundant, but it couldn't hurt!

Reply:
This doesn't parse because the nonterminal following the `in` keyword is ShiftExpression, but [ArrowFunction is an AssignmentExpression](https://tc39.es/ecma262/#sec-expressions). That's the same reason `yield` has to be parenthesized.

Reply:
We could add one, but between the test for YieldExpression (which uses ParenthesizedExpression) and the test for ShiftExpression (which does not), that seems more like a test for the wider grammar than for this proposal specifically. It makes me wonder if anyone has looked into precisely verifying expression precedence...